targettests/execution/state_preparation_vector.cpp

```diff
@@ -0,0 +1,159 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ %cpp_std --enable-mlir %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --enable-mlir --target quantinuum --emulate %s -o %t && %t | FileCheck %s
```

Comment:
Nit: `--enable-mlir` is not needed when specifying the hardware targets because the hardware targets' config files say `LIBRARY_MODE=false`.

Reply:
Thanks, will remove

Reply:
done